src/client-side-metrics/operation-metrics-collector-factory.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {MethodName, StreamingState} from './client-side-metrics-attributes';
+import {
+  ITabularApiSurface,
+  OperationMetricsCollector,
+} from './operation-metrics-collector';
+
+export class OperationMetricsCollectorFactory {
+  static createOperation(
+    methodName: MethodName,
+    streaming: StreamingState,
+    table: ITabularApiSurface,
+  ): OperationMetricsCollector | undefined {
+    if (table.bigtable.metricsEnabled) {
+      return new OperationMetricsCollector(table, methodName, streaming);
+    } else {
+      return;
```

Comment:
I think we should always write to the OperationMetricsCollector, but just do nothing if no handlers are attached. That makes the testing easier, since we don't have to worry about the two cases, optional values, etc. Let me know what you think

Reply:
It makes testing easier, but also means we run metrics collector code for no reason increasing latency.